extension/source/cells/CellCollection.ts

```diff
@@ -0,0 +1,206 @@
+import { EventEmitter } from 'events';
+
+import * as io from 'io-ts';
+
+import ExplicitAny from '../types/ExplicitAny';
+import ICell, { CellEmitter } from './ICell';
+import CellIterator from './CellIterator';
+import jsonHasChanged from './jsonHasChanged';
+import assert from '../helpers/assert';
+import IAsyncStorage from './IAsyncStorage';
+
+export default class CellCollection {
+  cells: Record<string, CollectionCell<ExplicitAny> | undefined> = {};
```

Comment:
We could, but I think that's unnecessary complexity actually.

Instead, if you have concrete types for the cells you want to use, I think it makes more sense to simply expose concrete cells like this:

```ts
const cc = new CellCollection(/* ... */);

const appCells = {
  foo: cc.Cell('foo', io.string, ''),
  bar: cc.Cell('bar', io.array(io.number), []),
};
```

This way you don't need to duplicate the type information, since you'll need to construct the runtime types at some point anyway.

Additionally, this allows for unrelated subsystems to share the same `CellCollection` without sharing any code. This is advantageous because `CellCollection` is usually some singleton-like persistent storage and `CellCollection` will use the same `CollectionCell` instance when using the same key which helps reduce overhead and checks the same type is being used if the `CollectionCell` is accessed by name in multiple places.